### PR TITLE
MRG: Rework CoregistrationUI fiducial I/O widgets & add save button

### PR DESCRIPTION
**Pull request by hoechenberger**
This is the last remaining bit of the #10237 rework, incorporating suggestions and feedback I've meanwhile received.

Closes #10237

I've removed the `_fiducials_file` traitlet as it's not needed anymore. I've moved its functionality to other methods.

https://user-images.githubusercontent.com/2046265/150946042-5513b999-ba82-460f-a427-49019e842037.mov

- Saving only allowed once fiducials are locked
- Reloding fiducials automatically locks fiducials

There's a bug related to the digpoints still showing even though the fiducials are unlocked. I've left an extensive comment in the source. This is probably something that needs to be solved in a separate PR unless @GuillaumeFavelier has an immediate idea? At any rate, I don't think this PR introduced this problem 😅

cc @agramfort

### CODE REVIEW

**Comment by GuillaumeFavelier** [comment]
I have multiple questions/remarks:
- Why `_fiducials_file` is not needed anymore? because if that is the case, we have to instantiate normally and not use `traitlets` at all (I think it's still created by it when I looked)
- I would avoid using lambda as callback (they do not work as you expect), try to use `partial` instead (maybe the bug is related to that)
- I recommend using the `_forward_widget_command()` function as much as possible, it checks that widget exists (instead of `_widgets[""]` directly)
- I saw we have another GUI API change with `selectable` in `_dock_add_label`, can we add a test for it?

**Comment by GuillaumeFavelier** [comment]
> Why's that?

When I designed the callbacks to set translation and rotation values for example, I used a `for`-loop to iterate over the modes (either 'translation' or 'rotation'). So one of the input parameter of the callback is a local variable corresponding to the current mode. I would expect the instantiated `lambda` to contain the current value as well but I noticed that it holds only the last (basically all the callbacks were only modifiying rotation). I dunno if this is clear but for me, it is an unexpected behaviour I observed with `lambda` and local variables. I personally fixed this issue by using `partial` so that's why I mean there is no 'surprise' with it.

In your case though, you do not use local variables but attributes of the class, it should be okay but I did not test your branch yet.

**Comment by hoechenberger** [comment]
> In your case though, you do not use local variables but attributes of the class, it should be okay but I did not test your branch yet.

Ah yes, I see what you mean. Yes this is specifically what I ensured before choosing to use `lambda`. But I will switch to `partial` just to be on the safe side and consistent with the rest of the code. Thanks for the explanation!

File: mne/gui/_coreg.py [hunk 1]
```diff
@@ -850,13 +893,29 @@ def _reset(self):
...
+        """Invoke a method of one or more widgets if the widgets exist.
+
+        Parameters
+        ----------
+        widget_names : str | array-like of str
+            The widget names to operate on.
+        method : str
+            The method to invoke.
+        values : object | array-like
+            The value(s) to pass to the method(s).
```
Comment (hoechenberger) [inline mne/gui/_coreg.py:906]:
```suggestion
            The value(s) to pass to the method.
```

File: mne/gui/_coreg.py [hunk 2]
```diff
@@ -466,7 +500,10 @@ def _subjects_dir_changed(self, change=None):
         # XXX: add coreg.set_subjects_dir
         self.coreg._subjects_dir = self._subjects_dir
         subjects = self._get_subjects()
-        self._subject = subjects[0]
+
+        if self._subject not in subjects:
```
Comment (hoechenberger) [inline mne/gui/_coreg.py:504]:
```suggestion
        if self._subject not in subjects:  # Just pick the first available one
```

**Comment by hoechenberger** [comment]
@GuillaumeFavelier I think this one is good to merge now! Basically what we cooked up together yesterday, plus some really minor changes

File: mne/gui/_coreg.py [hunk 3]
```diff
@@ -491,26 +538,21 @@ def _lock_fids_changed(self, change=None):
             self._scale_mode_changed()
             self._display_message()
             self._update_distance_estimation()
+            self._forward_widget_command('save_mri_fids', 'set_enabled', True)
```
Comment (GuillaumeFavelier) [inline mne/gui/_coreg.py:541]: I think we can add `'save_mri_fids'` to the `locked_widgets` list instead of calling `_forward_widget_command` directly.

**Review by GuillaumeFavelier — APPROVED** [review]
This is my only nitpick @hoechenberger 

About the bug, I will update #8833 :+1:  For now, the temporary workaround is to lock then unlock the fiducials :(

File: mne/gui/_coreg.py (hunk 3 above)
Comment (hoechenberger) [reply mne/gui/_coreg.py]: Done. 

Also see #10262 where I'm planning to address the "locking issue" more thoroughly.

**Pull request merged**